flow/project.py

```diff
@@ -245,7 +245,9 @@ def make_bundles(operations, size=None):
     return _make_bundles(operations, size)
 
 
-class JobOperation(object):
+@deprecated(
+    deprecated_in="0.11", removed_in="0.13", current_version=__version__)
+def JobOperation(id, name, job, cmd, directive=None):
```

Comment:
```suggestion
class JobOperation(_JobOperation):
```

This will allow the class name to remain the same while moving all the functionality to a private class.

Reply:
Never mind, we use the `deprecation` rather than `deprecated` Python package. The `deprecated` package offers a wrapper that works with classes as well rather than just methods and functions.

Reply:
This suggestion should work fine. Did you try it and find that it didn't? For instance:

```
>>> from deprecation import deprecated
>>> @deprecated()
... class Foo():
...     pass
...
>>> Foo()
<stdin>:1: DeprecatedWarning: Foo is deprecated
<__main__.Foo object at 0x7fdd6fcf9fa0>
```